### PR TITLE
Update strongnaming guidance

### PR DESCRIPTION
**Pull request by jkotas**
## Summary

Follow up on https://github.com/dotnet/aspire/pull/986#discussion_r1410939814

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/standard/library-guidance/strong-naming.md](https://github.com/dotnet/docs/blob/b37ee6d44a450fc53db8c615ac58d3b0eb332b39/docs/standard/library-guidance/strong-naming.md) | [Strong naming](https://review.learn.microsoft.com/en-us/dotnet/standard/library-guidance/strong-naming?branch=pr-en-us-38537) |

### CODE REVIEW

**Review by JamesNK — APPROVED** [review]

File: docs/standard/library-guidance/strong-naming.md [hunk 1]
```diff
@@ -24,17 +24,19 @@ The downside to strong naming is that .NET Framework on Windows enables strict l
...
+One important aspect of strong naming on .NET Framework is that it's viral: a strong named assembly can only reference other strong named assemblies. If your library isn't strong named, then you have excluded developers who are building an application or library that needs strong naming from using it on .NET Framework.
 
-The benefits of strong naming are:
+The benefits of strong naming on .NET Framework are:
 
 1. The assembly can be referenced and used by other strong-named assemblies.
 2. The assembly can be stored in the Global Assembly Cache (GAC).
 3. The assembly can be loaded side by side with other versions of the assembly. Side-by-side assembly loading is commonly required by applications with plug-in architectures.
 
+Strong naming has no benefits on .NET Core/5+.
```
Comment (danmoseley) [inline docs/standard/library-guidance/strong-naming.md:35]: No benefit, or also actually no effect?

**Review by danmoseley — APPROVED** [review]

File: docs/standard/library-guidance/strong-naming.md (hunk 1 above)
Comment (eerhardt) [reply docs/standard/library-guidance/strong-naming.md:35]: If you add/remove/change strong name keys on a library is that still a binary breaking change in .NET Core/5+? If it is, then strong naming still has an effect.

Comment (jkotas) [reply docs/standard/library-guidance/strong-naming.md:35]: > If you add/remove/change strong name keys on a library is that still a binary breaking change in .NET Core/5+? 

The runtime is going to be fine, but some of the build tools are likely going to choke on it. So yes - it is still a breaking change.

File: docs/standard/library-guidance/strong-naming.md [hunk 2]
```diff
@@ -24,17 +24,19 @@ The downside to strong naming is that .NET Framework on Windows enables strict l
 
 When .NET developers complain about strong naming, what they're usually complaining about is strict assembly loading. Fortunately, this issue is isolated to .NET Framework. .NET 5+, .NET Core, Xamarin, UWP, and most other .NET implementations don't have strict assembly loading, which is the main downside of strong naming.
 
-One important aspect of strong naming is that it's viral: a strong named assembly can only reference other strong named assemblies. If your library isn't strong named, then you have excluded developers who are building an application or library that needs strong naming from using it.
+One important aspect of strong naming on .NET Framework is that it's viral: a strong named assembly can only reference other strong named assemblies. If your library isn't strong named, then you have excluded developers who are building an application or library that needs strong naming from using it on .NET Framework.
```
Comment (gewarren) [inline docs/standard/library-guidance/strong-naming.md:27]:
```suggestion
One important aspect of strong naming on .NET Framework is that it's viral: a strong-named assembly can only reference other strong-named assemblies. If your library isn't strong named, then .NET Framework apps and libraries that need strong naming can't use it.
```

**Review by gewarren — APPROVED** [review]

File: docs/standard/library-guidance/strong-naming.md [hunk 3]
```diff
@@ -24,17 +24,19 @@ The downside to strong naming is that .NET Framework on Windows enables strict l
...
 1. The assembly can be referenced and used by other strong-named assemblies.
 2. The assembly can be stored in the Global Assembly Cache (GAC).
 3. The assembly can be loaded side by side with other versions of the assembly. Side-by-side assembly loading is commonly required by applications with plug-in architectures.
 
+Strong naming has no benefits on .NET Core/5+.
+
 ## Create strong named .NET libraries
 
-You should strong name your open-source .NET libraries. Strong naming an assembly ensures the most people can use it, and strict assembly loading only affects .NET Framework.
+You should strong name your open-source .NET libraries if their targets include .NET Framework or .NET Standard. Strong naming is not required for libraries that target .NET Core/5+ only.
```
[the remark this answers is not in the document]
Comment (eerhardt) [reply docs/standard/library-guidance/strong-naming.md:39]: Its the other way around (strong named assemblies referencing non-strong named assemblies get a warning). But I agree saying this warning is fine to suppress would be nice.

File: docs/standard/library-guidance/strong-naming.md [hunk 4]
```diff
@@ -24,17 +24,19 @@ The downside to strong naming is that .NET Framework on Windows enables strict l
...
+One important aspect of strong naming on .NET Framework is that it's viral: a strong-named assembly can only reference other strong-named assemblies. If your library isn't strong named, then .NET Framework apps and libraries that need strong naming can't use it.
 
-The benefits of strong naming are:
+The benefits of strong naming on .NET Framework are:
 
 1. The assembly can be referenced and used by other strong-named assemblies.
 2. The assembly can be stored in the Global Assembly Cache (GAC).
 3. The assembly can be loaded side by side with other versions of the assembly. Side-by-side assembly loading is commonly required by applications with plug-in architectures.
 
+Strong naming has no benefits on .NET Core/5+.
```
Comment (jkotas) [inline docs/standard/library-guidance/strong-naming.md:35]:
```suggestion
Strong naming has no benefits on .NET Core/5+. C# compiler produces CS8002 warning for strong named assemblies referencing non-strong named assemblies. It is fine to suppress this warning for libraries that target .NET Core/5+ only.
```

File: docs/standard/library-guidance/strong-naming.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (jkotas) [reply docs/standard/library-guidance/strong-naming.md:39]: > Its the other way around (strong named assemblies referencing non-strong named assemblies get a warning). But I agree saying this warning is fine to suppress would be nice.

Fixed.

File: docs/standard/library-guidance/strong-naming.md [hunk 5]
```diff
@@ -24,17 +24,19 @@ The downside to strong naming is that .NET Framework on Windows enables strict l
...
+One important aspect of strong naming on .NET Framework is that it's viral: a strong-named assembly can only reference other strong-named assemblies. If your library isn't strong named, then .NET Framework apps and libraries that need strong naming can't use it.
 
-The benefits of strong naming are:
+The benefits of strong naming on .NET Framework are:
 
 1. The assembly can be referenced and used by other strong-named assemblies.
 2. The assembly can be stored in the Global Assembly Cache (GAC).
 3. The assembly can be loaded side by side with other versions of the assembly. Side-by-side assembly loading is commonly required by applications with plug-in architectures.
 
+Strong naming has no benefits on .NET Core/5+. C# compiler produces CS8002 warning for strong named assemblies referencing non-strong named assemblies. It is fine to suppress this warning for libraries that target .NET Core/5+ only.
```
Comment (DamianEdwards) [inline docs/standard/library-guidance/strong-naming.md:35]:
```suggestion
Strong naming has no benefits on .NET Core/5+. C# compiler produces CS8002 warning for strong-named assemblies referencing non-strong named assemblies. It is fine to suppress this warning for libraries that target .NET Core/5+ only.
```

**Review by DamianEdwards — APPROVED** [review]

**Review by eerhardt — APPROVED** [review]
Thanks for updating this for the latest recommendations!

**Review by DamianEdwards — APPROVED** [review]

**Comment by terrajobst** [comment]
I'm late tot he party but I'd like to add that I like the changes.

**Pull request merged**